### ISSUE TITLE
Cannot uninstall package without url

### ISSUE THREAD

**Issue by scheibelp** [spack/spack#8173]
### Summary

See: https://github.com/spack/spack/pull/8066#discussion_r189062850

If you install a package and remove the version from it's definition. In some cases uninstalling it will produce a `NoURLError`. This does not always occur (the link mentions a case where it does).

### Steps to reproduce the issue

```console
$ spack install foo@2.1
# Remove version 2.1 from foo's package definition
$ spack uninstall foo@2.1
```

### Information on your system

This should be repeatable across all platforms without any special configuration.

**Comment by adamjstewart** [comment]
Possibly related to #98, #3042, #7971, and #8046. Also see #3161.

**Comment by jgalarowicz** [comment]
@scheibelp I was playing with my case and the problem goes away when I add a url line:
```
    homepage = "http://sourceforge.net/p/cbtf/wiki/Home"
    url='https://github.com/OpenSpeedShop/cbtf/wiki'

    # Use when the git repository is available
    version('1.9.1', branch='master',
            git='https://github.com/OpenSpeedShop/cbtf.git')
```
So, for my case, it might be my fault for not having the url line.

**Comment by scheibelp** [comment]
IMO you shouldn't have to add a URL for Spack to remove non-existent versions.

**Comment by adamjstewart** [comment]
Can you see if this is fixed by #3161?
